add labels to realm import jobs

KyriosGN0 opened the pull request:
<!---
Please read https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

closes #40799

KyriosGN0 committed:
add labels to realm import jobs

Signed-off-by: AvivGuiser <avivguiser@gmail.com>

KyriosGN0 committed:
add tests to check for labels in realm import jobs

Signed-off-by: AvivGuiser <avivguiser@gmail.com>

KyriosGN0 committed:
push labels from the realm import spec to the jobBuilder

Signed-off-by: AvivGuiser <avivguiser@gmail.com>

KyriosGN0 committed:
push labels from the realm import spec (and not metadata)  to the jobBuilder

Signed-off-by: AvivGuiser <avivguiser@gmail.com>

KyriosGN0 committed:
push labels from the realm import spec with addToLabels function in jobBuilder

Signed-off-by: AvivGuiser <avivguiser@gmail.com>

KyriosGN0 committed:
remove incorrect assert about job containing keycloak label

Signed-off-by: AvivGuiser <avivguiser@gmail.com>

shawkins commented:
Thank you for the changes @KyriosGN0 

A couple of thoughts:
- it would be good to add our default labels last to prevent overwriting built-in ones.
- do you anticipate using different labels for every import job? If not, there are also the import (and update) specs on the KeycloakSpec that could be used to define labels for their respective jobs.

KyriosGN0 committed:
insert default labels last to realm import job, add labels to updateJob

Signed-off-by: AvivGuiser <avivguiser@gmail.com>

KyriosGN0 committed:
change WithLabels to AddToLabels to not fail on null values

Signed-off-by: AvivGuiser <avivguiser@gmail.com>

KyriosGN0 committed:
use Optional in case labels is empty for updateJob and realm import job

Signed-off-by: AvivGuiser <avivguiser@gmail.com>

KyriosGN0 committed:
create optionalSpec for updateJob and RealmImport Job and use it to populate the labels map

Signed-off-by: AvivGuiser <avivguiser@gmail.com>

# Conflicts:
#	operator/src/main/java/org/keycloak/operator/controllers/KeycloakUpdateJobDependentResource.java

KyriosGN0 committed:
create optionalSpec for updateJob and RealmImport Job and use it to populate the labels map

Signed-off-by: AvivGuiser <avivguiser@gmail.com>

KyriosGN0 committed:
fix test of updateJob and labels typo

Signed-off-by: AvivGuiser <avivguiser@gmail.com>

shawkins reviewed (approved):
LGTM, thank you @KyriosGN0

vmuzikar reviewed (approved):
Thank you for the contribution, LGTM.

ahus1 commented:
@devshrm - if you look at the parent issue, you'll see that the issue is tagged `26.5.0`. That feature release is due for January 2026. In the meantime, there will be patch releases for 26.4, but those only include bug fixes, not enhancements.
